packages/core/src/sync/source.ts

```diff
@@ -77,3 +78,29 @@ export const isAddressFactory = (
     return false;
   return Array.isArray(address) ? isAddressFactory(address[0]) : true;
 };
+
+export const getChildAddress = ({
+  log,
+  factory,
+}: { log: SyncLog; factory: Factory }): Address => {
+  if (factory.childAddressLocation.startsWith("offset")) {
+    const childAddressOffset = Number(
+      factory.childAddressLocation.substring(6),
+    );
+    const start = 2 + 12 * 2 + childAddressOffset * 2;
+    const length = 20 * 2;
+
+    return `0x${log.data.substring(start, start + length)}`;
+  } else {
+    const start = 2 + 12 * 2;
+    const length = 20 * 2;
+    const topicIndex =
+      factory.childAddressLocation === "topic1"
+        ? 1
+        : factory.childAddressLocation === "topic2"
+          ? 2
+          : 3;
+    log.topics;
```

Comment:
Typo?

Looks good to me, but while we're here I think it's worthwhile to add a couple unit tests for this, e.g. around casing